node/src/components/chain_synchronizer/operations.rs

```diff
@@ -737,30 +739,62 @@ where
             parent: Box::new(parent_header.clone()),
         })?;
 
-    // Ensure we have at least one peer to which we're bidirectionally connected before trying to
-    // fetch the data.
-    let mut peers = vec![];
-    for _ in 0..ctx.config.max_retries_while_not_connected() {
-        peers = get_filtered_fully_connected_peers(ctx).await;
-        if !peers.is_empty() {
-            break;
+    for _ in 0..=1 {
+        let peers = prepare_peers_applicable_for_block_fetch(ctx).await;
+        let peer_count = peers.len();
+        let maybe_item = try_fetch_block_or_block_header_by_height(
+            peers.clone(),
+            ctx,
+            height,
+            parent_header,
+            key_block_info,
+        )
+        .await?;
+        match maybe_item {
+            Some(item) => {
+                check_block_version_is_not_greater_than_current(
+                    item.header(),
+                    ctx.config.protocol_version(),
+                )?;
+                check_block_version_is_not_lower_than_parent(item.header(), parent_header)?;
```

Comment:
We inlined these in a recent mob session and removed the test for `check_block_version`, so I think we should do that here too.

Reply:
Inlined in 0b7f912834fb322bac951d8a4c2df8fffbdc1756. I also removed the test.
I'll solve potential conflicts after #3214 merges.

Reply:
Conflicts solved here: https://github.com/casper-network/casper-node/pull/3210/commits/b7add644c8e39d2b36beb2f35aff5e864f382416